streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -346,8 +348,14 @@ public void writeAsText(final String filePath,
         return branchChildren;
     }
 
+    
+    @Override 
+    public KStream<K, V> merge(KStream<K, V> stream) {
+        return this.merge(this.builder, (KStream<K, V>) this, stream);
+    }
+    
     public static <K, V> KStream<K, V> merge(final InternalStreamsBuilder builder,
-                                             final KStream<K, V>[] streams) {
+                                             final KStream<K, V> ... streams) {
```

Comment:
remove `vararg` here -- we should only take a single `KStream`

-> I assume this "breaks" deprecated `KStreamBuilder#merge` and I would fix it there to call `KStreamImpl.merge()` in a loop for all given parameters.